File: Modules/ServerTypes/WorkFlow/ActionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ActionRow } from "./ActionRow";

export interface ActionColumns {
    Id: Column<ActionRow>;
    Name: Column<ActionRow>;
    UpdateDate: Column<ActionRow>;
}

export class ActionColumns extends ColumnsBase<ActionRow> {
    static readonly columnsKey = 'WorkFlow.Action';
    static readonly Fields = fieldsProxy<ActionColumns>();
}

File: Modules/WorkFlow/Action/ActionColumns.cs
namespace WorkFlow.Columns;

[ColumnsScript("WorkFlow.Action")]
[BasedOnRow(typeof(ActionRow), CheckNames = true)]
public class ActionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink, Width(200)]
    public string Name { get; set; }
    [Width(200)]
    public DateTime UpdateDate { get; set; }
}